Automattic/jetpack
# Documentation: Update formatting and structure for Development environment document

#### Changes proposed in this Pull Request:

* This PR introduces an update to the Development Environment setup document.
* Updated:
 * Formatting
  * Some things were a bit hard to follow when I set up my environment
 * Organization
  * Some points are now grouped together 
  * Tried to order the document in more step-by-step flow, compared to more "group of notes" style

#### Testing instructions:

* Open `docs/development-environment.md`
* Read the document and see if it makes sense and does it look better than it did before

#### Proposed changelog entry for your changes:

* Not sure if this will need a changelog entry as it doesn't affect the plugin itself.
